.github/actions/composite/setupGitForOSBotifyApp/action.yml

```diff
@@ -24,6 +24,21 @@ outputs:
 runs:
   using: composite
   steps:
+    - name: Check if gpg encrypted private key is present
+      id: key_check
+      shell: bash
+      run: |
+        if [[ -f .github/workflows/OSBotify-private-key.asc.gpg ]]; then
+          echo "::set-output name=key_exists::true"
+        fi
+  
+    - name: Checkout
+      uses: actions/checkout@v3
+      if: steps.key_check.outputs.key_exists != 'true'
+      with:
+        sparse-checkout: |
+          .github
```

Comment:
@roryabraham do you know if we need to specify the github token here? Does the default `github.token` not work?

Reply:
Default token for `actions/checkout` works 👍🏼 

We've never used this `sparse-checkout` before, but it's a cool way to speed up checkout. will start using everywhere 😈